### PR TITLE
Clickable InlayHint for Java type

### LINKED ISSUE #702: Clickable InlayHint for Java type
Given this Qute template:

```
{#for item in items}
	{item.price}
{/for}
```

Inlay hint display the item Java class type:

image

It should be nice to give the capability to click on this inlay hint to open the Java class.

### PR DESCRIPTION
**Pull request by angelozerr**
Clickable InlayHint for Java type

Fixes #702

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

**Comment by fbricon** [comment]
InlyHint => InlayHint

**Comment by angelozerr** [comment]
This PR requires https://github.com/redhat-developer/vscode-quarkus/pull/533

**Comment by fbricon** [comment]
> This PR requires [redhat-developer/vscode-quarkus#533](https://github.com/redhat-developer/vscode-quarkus/pull/533)

No it doesn't. It's the other way around

**Comment by angelozerr** [comment]
Here a demo:

ClickableInlayHintDemo

**Comment by fbricon** [comment]
can you not underline `:`, I find it disturbing

**Comment by angelozerr** [comment]
> can you not underline :, I find it disturbing

It is the vscode behavior. I dont know if it is possible to customize this behavior.

**Comment by angelozerr** [comment]
>can you not underline `:`, I find it disturbing

I was wrong, it is possible by generating two InlayHintLabelPart 

 * one for  `:` without command (not clickable)
 * one for Java type (ex : Item) which is clickable

image

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/inlayhint/InlayHintASTVistor.java [hunk 1]
```diff
@@ -239,8 +250,31 @@ private static CompletableFuture<ResolvedJavaTypeInfo> getJavaType(Parameter par
...
-	private static void updateJavaType(InlayHint hint, ResolvedJavaTypeInfo javaType) {
-		String type = javaType != null ? javaType.getJavaElementSimpleType() : "?";
-		hint.setLabel(Either.forLeft(":" + type));
+	private void updateJavaType(InlayHint hint, ResolvedJavaTypeInfo javaType, String projectUri) {
+		String type = getLabel(javaType);
+		if (javaType != null && canSupportJavaDefinition) {
+			// first part : ":"
+			InlayHintLabelPart separatorPart = new InlayHintLabelPart(":");
+			// second part label : clickable Java type (ex : String)
+			InlayHintLabelPart javaTypePart = new InlayHintLabelPart(type.toString());
```
Comment (datho7561) [inline qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/inlayhint/InlayHintASTVistor.java:259]: `type` is already a string (the label string for `javaType`), so there is not need to call toString()

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/inlayhint/InlayHintASTVistor.java (hunk 1 above)
Comment (angelozerr) [reply qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/inlayhint/InlayHintASTVistor.java]: Good catch, fixed

**Review by datho7561 — APPROVED** [review]
Looks good to me! Well done, Angelo!

**Pull request merged**